### PR TITLE
Simplify sending HTTP requests 

### PR DESCRIPTION
**Pull request by stefanbirkner**
Avoid duplication.

### CODE REVIEW

File: src/main/java/com/saucelabs/rdc/RdcAppiumSuite.java [hunk 1]
```diff
@@ -205,44 +198,29 @@ public void run(RunNotifier notifier) {
...
-		WebTarget target = client
-			.path("suites").path(Long.toString(suiteId))
-			.path("reports")
-			.path("start");
+		Request request = baseRequest
+			.path("suites/" + suiteId + "/reports/start");
 		if (appId.isPresent()) {
-			String appIdAsString = Long.toString(appId.getAsLong());
-			target = target.queryParam("appId", appIdAsString);
+			request = request.queryParam("appId", appId.getAsLong());
```
Comment (abdullahawara) [inline src/main/java/com/saucelabs/rdc/RdcAppiumSuite.java:215]: Is this tested? I'm always worried by this kind of changes

File: src/main/java/com/saucelabs/rdc/helper/Request.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+			return request.apply(invocationBuilder(target));
+		} finally {
+			client.close();
+		}
+	}
+
+	private Invocation.Builder invocationBuilder(WebTarget target) {
+		target = target.path("/rest/v2/appium" + path);
+		for (Map.Entry<String, Object> queryParam: queryParams.entrySet()) {
+			target = target.queryParam(
```
Comment (abdullahawara) [inline src/main/java/com/saucelabs/rdc/helper/Request.java:87]: Do we really need to re-assign the `request` after we add a query param to it?

File: src/main/java/com/saucelabs/rdc/RdcAppiumSuite.java (hunk 1 above)
Comment (stefanbirkner) [reply src/main/java/com/saucelabs/rdc/RdcAppiumSuite.java:215]: It is covered by the test `appIdIsUpdatedWhenPresentAsEnvironmentVariable` in [RdcAppiumSuiteTest](https://github.com/saucelabs/rdc-appium-junit4/blob/master/src/test/java/com/saucelabs/rdc/RdcAppiumSuiteTest.java).

File: src/main/java/com/saucelabs/rdc/helper/Request.java (hunk 2 above)
Comment (stefanbirkner) [reply src/main/java/com/saucelabs/rdc/helper/Request.java:87]: Yes. `WebTarget` is immutable. `WebTarget#path` and the other methods create new `WebTarget` objects.

**Review by abdullahawara — APPROVED** [review]

**Review by gerritschultz — APPROVED** [review]

**Pull request merged**